pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -411,7 +411,7 @@ func (ku *KubeUtil) setupKubeletApiEndpoint() error {
 			log.Debugf("Kubelet endpoint is: %s", ku.kubeletApiEndpoint)
 			return nil
 		}
-		if code != http.StatusUnauthorized {
+		if code < 400 && code >= 500 {
```

Comment:
There is now a codepath where this log will occur even when `httpsUrlErr == nil`.

```
log.Debugf("Cannot query %s%s: %s", ku.kubeletApiEndpoint, kubeletPodPath, httpsUrlErr)
```

Is it also possible to maybe print the status code we did get in the warning. It might be nice to know if we are always getting `403` or something else when this happens.